log-viewer/src/main/java/com/logviewer/data2/Log.java

```diff
@@ -170,6 +167,25 @@ private Path decompressAndCopyGZipFile() throws IOException {
         return tempFile;
     }
 
+    private Path decompressAndCopyZipFile() throws IOException {
+        Path tempFile = Files.createTempFile(Utils.getTempDir(), "unpacked-zip-", "-" + file.getName(file.getNameCount() - 1) + ".tmp");
+        tempFile.toFile().deleteOnExit();
+
+        try (ZipInputStream zis = new ZipInputStream(Files.newInputStream(file))) {
+            ZipEntry zipEntry = zis.getNextEntry();
+            while(zipEntry != null){
```

Comment:
You expect one file in the archive, let's throw an exception if the archive contains more than one file. 
```
throw new IOException("ZIP archive contains more than one file. Log-viewer supports zip archives with one file only")
```